### PR TITLE
Adding/Fixing code quality tools

### PR DESCRIPTION
**Pull request by mcassiano**
- [x] Fixed JaCoCo and added support for all modules
- [x] Added aggregated test report (report is able to tell the code coverage for all sub projects)
- [x] Added KtLint for Checkstyle and Linting
- [x] Added Detekt for common mistakes and inconsistencies

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,29 +1,36 @@
...
+}
+
+apply plugin: "jacoco"
+
+jacoco {
+    toolVersion = "0.8.0"
+}
+
 allprojects {
     repositories {
```
Comment (cs-yasmin-pereira) [inline build.gradle]: Maybe if the block of allProjects was before the jacoco plugin

File: build.gradle [hunk 2]
```diff
@@ -1,29 +1,36 @@
...
         google()
         jcenter()
     }
 }
```
Comment (cs-yasmin-pereira) [inline build.gradle:24]: Maybe if the block of allProjects was before the jacoco plugin

File: build.gradle [hunk 3]
```diff
@@ -45,4 +52,37 @@ ext {
     roboeletricVersion = "3.6.1"
     canarinhoVersion = "1.2.0"
     lottieVersion = "2.5.0-rc1"
-}
\ No newline at end of file
+    ktlintVersion = "0.19.0"
+    detektVersion = "1.0.0.RC6-3"
+}
```
Comment (cs-yasmin-pereira) [inline build.gradle]: Move ext to the end of the file

File: build.gradle (hunk 2 above)
Comment (mcassiano) [reply build.gradle:24]: Done

File: build.gradle (hunk 3 above)
Comment (mcassiano) [reply build.gradle]: Done

**Pull request merged**